flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -155,55 +199,79 @@ public GetRecordsResult getRecords(String shardIterator, int maxRecordsToGet) {
 	 * @param startingSeqNum the sequence number that the iterator will start from
 	 * @return the shard iterator
 	 */
+	@Override
 	public String getShardIterator(KinesisStreamShard shard, String shardIteratorType, String startingSeqNum) {
-		return kinesisClient.getShardIterator(shard.getStreamName(), shard.getShardId(), shardIteratorType, startingSeqNum).getShardIterator();
+		return kinesisClient.getShardIterator(shard.getStreamName(), shard.getShard().getShardId(), shardIteratorType, startingSeqNum).getShardIterator();
```

Comment:
Yes, I think we should have retries there as well. In particular during start up, many parallel Flink instances will query AWS for the shard iterator.